src/Task/Runner/TaskRunner.php

```diff
@@ -59,68 +70,72 @@ public function __construct(
      */
     public function runTasks()
     {
-        $executions = $this->taskExecutionRepository->findScheduled();
-
-        foreach ($executions as $execution) {
-            // this find is neccesary if the storage because the storage could be
-            // invalid (clear in doctrine) after handling an execution.
-            $execution = $this->taskExecutionRepository->findByUuid($execution->getUuid());
-
-            $start = microtime(true);
-            $execution->setStartTime(new \DateTime());
-            $execution->setStatus(TaskStatus::RUNNING);
-            $this->taskExecutionRepository->save($execution);
-
-            try {
-                $result = $this->handle($execution);
-
-                // this find is neccesary if the storage because the storage could be
-                // invalid (clear in doctrine) after handling an execution.
-                $execution = $this->taskExecutionRepository->findByUuid($execution->getUuid());
-                $execution->setStatus(TaskStatus::COMPLETED);
-                $execution->setResult($result);
-
-                $this->eventDispatcher->dispatch(
-                    Events::TASK_PASSED,
-                    new TaskExecutionEvent($execution->getTask(), $execution)
-                );
-            } catch (\Exception $ex) {
-                // this find is neccesary if the storage because the storage could be
-                // invalid (clear in doctrine) after handling an execution.
-                $execution = $this->taskExecutionRepository->findByUuid($execution->getUuid());
-                $execution->setException($ex->__toString());
-                $execution->setStatus(TaskStatus::FAILED);
-
-                $this->eventDispatcher->dispatch(
-                    Events::TASK_FAILED,
-                    new TaskExecutionEvent($execution->getTask(), $execution)
-                );
+        $runTime = new \DateTime();
+
+        while ($execution = $this->taskExecutionRepository->findNextScheduled($runTime)) {
```

Comment:
Is performance a concern here? You are multiplying the database requests this way. Can't you still load all of them at once and keep them in memory before executing them?